src/v/storage/disk_log_impl.cc

```diff
@@ -784,18 +797,7 @@ ss::future<> disk_log_impl::gc(compaction_config cfg) {
           config().ntp());
         co_return;
     }
-    if (cfg.max_bytes) {
-        size_t max = cfg.max_bytes.value();
-        vlog(
-          gclog.debug,
-          "[{}] retention max bytes: {}, current partition size: {}",
-          config().ntp(),
-          max,
-          _probe.partition_size());
-        if (!_segs.empty() && _probe.partition_size() > max) {
-            co_return co_await garbage_collect_max_partition_size(cfg);
-        }
-    }
+    co_await garbage_collect_max_partition_size(cfg);
```

Comment:
Rather than potentially GCing twice, could we determine both offsets up front, and then call `garbage_collect_segments()` on which ever is higher?

Reply:
yeh, i thought about. we could do that and i'm neutral about it. i'm not sure it will offer any performance advantage since both translations to offsets have to be computed, and then garbage_collect_segments will return almost immediately in the second case if the first segment is no longer relevant for the target offset.

Reply:
Agreed, not much functional gain -- it's just less surprising to see looking at this fresh.